src/main/java/org/candlepin/subscriptions/security/SecurityConfig.java

```diff
@@ -42,144 +44,195 @@
+                .anonymous() // Creates an anonymous user if no header is present at all. Prevents NPEs
+                .and()
+                .authorizeRequests()
+                    .antMatchers("/**/openapi.*", "/**/version", "/api-docs/**", "/webjars/**").permitAll()
+                    // ingress security uses server settings (require ssl cert auth), so permit all here
+                    .antMatchers(String.format("/%s/ingress/**", apiPath)).permitAll()
+                    .requestMatchers(EndpointRequest.toAnyEndpoint()).permitAll()
+                    .anyRequest().authenticated();
+        }
     }
 
-    @Bean
-    public AccessDeniedHandler restAccessDeniedHandler() {
-        return new RestAccessDeniedHandler(mapper);
+    /**
+     * Security configuration for Jolokia Actuator.  Jolokia has GET endpoints that can affect the state of
+     * the application.  We need to protect these endpoints from CSRF attacks.  Normally GET requests are
+     * exempted from CSRF restrictions, so we need to create a special security configuration and use a
+     * special AntiCsrfFilter to guard GET requests on the /actuator/jolokia context.
+     *
+     * This security configuration will be loaded before the normal configuration.
+     */
+    @Order(1)
+    @Configuration
+    @Import(ApiConfiguration.class)
+    public static class JolokiaActuatorConfiguration extends WebSecurityConfigurerAdapter {
+        @Autowired private ApplicationProperties appProps;
+        @Autowired private ConfigurableEnvironment env;
+        @Autowired protected ObjectMapper mapper;
+
+        // NOTE: intentionally *not* annotated with @Bean; @Bean causes an extra use as an application filter
+        public AntiCsrfFilter getVerbIncludingAntiCsrfFilter(ApplicationProperties appProps,
+            ConfigurableEnvironment env) {
+            return new GetVerbIncludingAntiCsrfFilter(appProps, env);
+        }
+
+
+        // NOTE: intentionally *not* annotated w/ @Bean; @Bean causes an *extra* use as an application filter
+        public IdentityHeaderAuthenticationFilter identityHeaderAuthenticationFilter() throws Exception {
+            IdentityHeaderAuthenticationFilter filter = new IdentityHeaderAuthenticationFilter(mapper);
+            filter.setCheckForPrincipalChanges(true);
+            filter.setAuthenticationManager(authenticationManager());
+            filter.setAuthenticationFailureHandler(new IdentityHeaderAuthenticationFailureHandler(mapper));
+            filter.setContinueFilterChainOnUnsuccessfulAuthentication(false);
+            return filter;
+        }
+
+        @Override
+        protected void configure(HttpSecurity http) throws Exception {
+            // See https://docs.spring.io/spring-security/site/docs/current/reference/html5/#ns-custom-filters
+            // for list of filters and their order
+            http
+                .requestMatcher(EndpointRequest.to("jolokia")) // restrict this config to Jolokia
```

Comment:
Unfortunately, there is one more subpath to match. Hawtio creates a [forwarding controller](https://github.com/hawtio/hawtio/blob/c9d65be7081072ac836218c51282c91b99dacc4c/platforms/springboot/src/main/java/io/hawt/springboot/HawtioManagementConfiguration.java#L308) from `/actuator/hawtio/jolokia` (and subpaths) to `/actuator/jolokia`, so a couple quick tests that need to be addressed:
```sh
for subpath in actuator/jolokia actuator/jolokia/list actuator/hawtio/jolokia actuator/hawtio/jolokia/list; do
  test "$(curl -s http://localhost:8080/$subpath -i | head -n 1 | cut -d' ' -f2)" != "401" && echo "[FAILED] - curl -s http://localhost:8080/$subpath"
done
```
shows:
```
[FAILED] - curl -s http://localhost:8080/actuator/hawtio/jolokia
[FAILED] - curl -s http://localhost:8080/actuator/hawtio/jolokia/list
```

Reply:
Fixed:

Thanks for the handy shell test loop!
```
% for subpath in actuator/jolokia actuator/jolokia/list actuator/hawtio/jolokia actuator/hawtio/jolokia/list; do
  curl -H "Origin: https://localhost:8080" -s http://localhost:8080/$subpath -i | head -n 1 | cut -d' ' -f2 
done
200
200
200
200

% for subpath in actuator/jolokia actuator/jolokia/list actuator/hawtio/jolokia actuator/hawtio/jolokia/list; do
  curl -s http://localhost:8080/$subpath -i | head -n 1 | cut -d' ' -f2 
done
401
401
401
401
```